main/main.c

```diff
@@ -142,10 +191,27 @@ run_signal_handler(void)
 int
 main(int argc, char **argv)
 {
+	struct args args = {
+		/* Defaults. */
+		.lua_base_dir = LUA_BASE_DIR,
+		.gatekeeper_config_file = GATEKEEPER_CONFIG_FILE,
+	};
+
 	int ret = rte_eal_init(argc, argv);
 	if (ret < 0)
 		rte_exit(EXIT_FAILURE, "Error with EAL initialization\n");
 
+	argc -= ret;
+	argv += ret;
+
+	/* Parse Gatekeeper arguments (after the EAL ones). */
+	ret = argp_parse(&argp, argc, argv, 0, NULL, &args);
+	if (ret != 0)
+		rte_exit(EXIT_FAILURE, "Invalid Gatekeeper parameters\n");
+
+	lua_base_dir = args.lua_base_dir;
+	gatekeeper_config_file = args.gatekeeper_config_file;
```

Comment:
Let's pass `args.lua_base_dir` and `args.gatekeeper_config_file` as parameters of `config_gatekeeper()` and get rid of the global variables `lua_base_dir` and `gatekeeper_config_file`.

I know that I've said that this patch was ready, but I couldn't resist this cute improvement once I saw it.